spec/code-block-generator-spec.coffee

```diff
@@ -51,66 +64,87 @@ describe 'Code block generator', ->
       codeBlocks = generator.makeAsciidocBlocks(languages)
       expect(codeBlocks).toHaveLength 3 # Number of blocks
       expect(codeBlocks[0]).toEqualJson
-        begin: '^\\[(source),\\p{Blank}*(?i:(javascript|js))(?:,([^\]]*))?\\]$'
-        beginCaptures:
-          0: name: 'support.asciidoc'
-          1: name: 'entity.name.function.asciidoc'
-          2: name: 'entity.name.type.asciidoc'
-          3:
-            name: 'markup.meta.attribute-list.asciidoc'
-            patterns: [
-              include: '#attribute-reference'
-            ]
+        name: "markup.code.js.asciidoc"
+        begin: "(?=(?>(?:^\\[(source)(?:,|#)\\p{Blank}*(?i:(javascript|js))((?:,|#)[^\\]]+)*\\]$)))"
         patterns: [
-          name: 'markup.code.js.asciidoc'
+          match: "^\\[(source)(?:,|#)\\p{Blank}*(?i:(javascript|js))((?:,|#)([^,\\]]+))*\\]$"
+          captures:
+            0:
+              name: 'markup.heading.asciidoc'
+              patterns: [
+                include: '#block-attribute-inner'
+              ]
+        ,
+          include: '#inlines'
+        ,
+          include: '#block-title'
+        ,
+          comment: 'listing block'
           begin: '^(-{4,})\\s*$'
-          beginCaptures:
-            0: name: 'support.asciidoc'
-          contentName: 'source.embedded.js'
+          contentName: "source.embedded.js"
           patterns: [
             include: '#block-callout'
           ,
-            include: 'source.js'
+            include: "source.js"
           ]
-          end: '^\\1*$'
-          endCaptures:
-            0: name: 'support.asciidoc'
+          end: '^(\\1)$'
+        ,
+          comment: 'open block'
+          begin: '^(-{2})\\s*$'
+          contentName: "source.embedded.js"
+          patterns: [
+            include: '#block-callout'
+          ,
+            include: "source.js"
+          ]
+          end: '^(\\1)$'
         ]
-        end: '(?<=----)[\\r\\n]+$'
+        end: '((?<=--)[\\r\\n]+$|^\\p{Blank}*$)'
 
     it 'should generate C++ code block', ->
       languages = [
         pattern: 'c(pp|\\+\\+)', type: 'source', code: 'cpp'
       ]
       codeBlocks = generator.makeAsciidocBlocks(languages)
       expect(codeBlocks).toHaveLength 3 # Number of blocks
+      # begin: '^\\[(source),\\p{Blank}*(?i:(c(pp|\\+\\+)))(?:,([^\]]*))?\\]$'
```

Comment:
I assume this comment line can be removed, as it seems to be more of a WIP annotation.